pallets/trustless-eth-bridge/substrate-app/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+        Burned(
+            SubNetworkId,
+            AssetIdOf<T>,
+            T::AccountId,
+            ParachainAccountId,
+            BalanceOf<T>,
+        ),
+        /// [network_id, asset_id, sender, recepient, amount]
+        Minted(
+            SubNetworkId,
+            AssetIdOf<T>,
+            ParachainAccountId,
+            T::AccountId,
+            BalanceOf<T>,
+        ),
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn asset_kind)]
+    pub(super) type AssetKinds<T: Config> =
+        StorageDoubleMap<_, Identity, SubNetworkId, Identity, AssetIdOf<T>, AssetKind, OptionQuery>;
+
+    #[pallet::error]
+    pub enum Error<T> {
+        TokenIsNotRegistered,
+        AppIsNotRegistered,
+        NotEnoughFunds,
+        InvalidNetwork,
+        TokenAlreadyRegistered,
+        AppAlreadyRegistered,
+        /// Call encoding failed.
+        CallEncodeFailed,
+        /// Amount must be > 0
+        WrongAmount,
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /*
+        Internal calls to be used from Ethereum side.
+        DON'T CHANGE ORDER
+         */
+
+        #[pallet::weight(<T as Config>::WeightInfo::mint())]
+        pub fn mint(
+            origin: OriginFor<T>,
+            asset_id: AssetIdOf<T>,
+            sender: ParachainAccountId,
+            recipient: T::AccountId,
+            amount: BalanceOf<T>,
+        ) -> DispatchResult {
+            let CallOriginOutput {
+                network_id,
+                message_id,
+                timestamp,
+                ..
+            } = T::CallOrigin::ensure_origin(origin.clone())?;
+
+            let asset_kind = AssetKinds::<T>::get(network_id, asset_id)
+                .ok_or(Error::<T>::TokenIsNotRegistered)?;
+
+            let bridge_account = Self::bridge_account()?;
+
+            ensure!(amount > 0, Error::<T>::WrongAmount);
+            match asset_kind {
+                AssetKind::Thischain => {
+                    assets::Pallet::<T>::transfer_from(
+                        &asset_id,
+                        &bridge_account,
+                        &recipient,
+                        amount,
+                    )?;
+                }
+                AssetKind::Sidechain => {
+                    assets::Pallet::<T>::mint_to(&asset_id, &bridge_account, &recipient, amount)?;
+                }
+            }
+            T::MessageStatusNotifier::inbound_request(
+                GenericNetworkId::Sub(network_id),
+                message_id,
+                GenericAccount::Parachain(sender.clone()),
+                GenericAccount::Sora(recipient.clone()),
+                asset_id,
+                amount,
+                timestamp,
+            );
+            Self::deposit_event(Event::Minted(
+                network_id, asset_id, sender, recipient, amount,
+            ));
+            Ok(())
+        }
+
+        /*
+        Common exstrinsics
+         */
```

Comment:
Minor: this comment style looks odd.
I think it's better to use:
`// regular comment`
or
`/*`
` * doxygen style comment`
` */`